openfe_gromacs/protocols/gromacs_md/md_methods.py

```diff
@@ -245,14 +245,18 @@ def get_filenames_em(self) -> dict[str, list[pathlib.Path]]:
             "cpt_em",
         ]
         dict_em = {}
+
         for file in file_keys:
+            try:
```

Comment:
`try/except` can be really expensive _if_ the exception gets caught.

i.e. in cases where you don't have a file, this will be expensive all the time, and all the other times the cost will be negligible.

Is there a way we can avoid this pattern?

Reply:
Changed this to just add another if requirement. It would return an empty list if there shouldn't be a file, so I changed that to return `None` when there's an empty list, though I guess we could also return an empty list, in case you think that's better!